backend/app/routers/listeners.py

```diff
@@ -374,10 +391,66 @@ async def edit_listener(
     raise HTTPException(status_code=404, detail=f"listener {listener_id} not found")
 
 
+@router.put("/{listener_id}/enable", response_model=EventListenerOut)
+async def enable_listener(
+    listener_id: str,
+    user_id=Depends(get_user),
+    allow: bool = Depends(ListenerAuthorization()),
+):
+    """Enable an Event Listener. Only admins can enable listeners.
+
+    Arguments:
+        listener_id -- UUID of the listener to be enabled
+    """
+    return await set_active_flag(listener_id, True)
+
+
+@router.put("/{listener_id}/disable", response_model=EventListenerOut)
+async def disable_listener(
+    listener_id: str,
+    user_id=Depends(get_user),
+    allow: bool = Depends(ListenerAuthorization()),
+):
+    """Disable an Event Listener. Only admins can enable listeners.
+
+    Arguments:
+        listener_id -- UUID of the listener to be enabled
+    """
+    return await set_active_flag(listener_id, False)
+
+
+@router.put("/{listener_id}/toggle", response_model=EventListenerOut)
+async def set_active_flag(
+    listener_id: str,
+    active: Optional[bool] = None,
+    allow: bool = Depends(ListenerAuthorization()),
+):
+    """Toggle the active flag of an Event Listener. Only admins can enable/disbale listeners.
+
+    Arguments:
+        listener_id -- UUID of the listener to be enabled
+    """
+    listener = await EventListenerDB.find_one(
+        EventListenerDB.id == ObjectId(listener_id)
+    )
+    if listener:
```

Comment:
in the dependency class `ListenerAuthorization`, You might be already checking the id exist?

Reply:
I don't if admin and admin_mode both are true